sdk/storage/blobs/blob_client.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+namespace azure { namespace storage { namespace blobs {
+
+class blob_client
+{
+public:
+    // connection string
+    blob_client(const std::string& connection_string, std::string container_name, std::string blob_name, blob_client_options options);
+
+    // shared key auth 
+    blob_client(azure::core::http::uri uri, azure::storage::common::storage_shared_key_credential credential, blob_client_options options);
+
+    // token auth
+    blob_client(azure::core::http::uri uri, azure::core::token_credential credential, blob_client_options options);
+
+    // anonymous/SAS/customized pipeline auth
+    blob_client(azure::core::http::uri uri, blob_client_options options);
+
+    bool is_snapshot() const;
+
+    azure::core::http::response<void> delete_(const delete_blob_options& options = delete_blob_options());
+
+    azure::core::http::response<bool> exists(const blob_exists_options& options = blob_exists_options());
+
+    azure::core::http::response<blob_properties> get_properties(const get_blob_properties_options& options = get_blob_properties_options());
```

Comment:
`get_blob_properties` and `get_blob_metadata` instead of omitting blob?

Reply:
I don't think we need to repeat *blob* again in API names. If so, how about `create_blob`, `delete_blob`, `create_blob_container`, `delete_blob_container`, after all we need to keep consistent.